fs/move.py

```diff
@@ -64,6 +64,10 @@ def move_file(
     with manage_fs(src_fs, writeable=True) as _src_fs:
         with manage_fs(dst_fs, writeable=True, create=True) as _dst_fs:
             if _src_fs is _dst_fs:
+                # Exit early if source and destination are the same file
+                if normpath(src_path) == normpath(dst_path):
+                    return
```

Comment:
@tfeldmann Is this part essential for _this_ PR, or is it (and the related tests) best left for #547 ?

Reply:
@lurch : this should be fine if I merge #547 first, after the rebase this should not appear as a change anymore I think.

Reply:
> @tfeldmann Is this part essential for _this_ PR, or is it (and the related tests) best left for #547 ?

Depends, because we use `overwrite=True` not having this would introduce a bug in this PR.

Reply:
We should use `validatepath` here, too. I'm fixing this up right now.

Reply:
Update: I found this check is no longer needed because moving onto itself is now handled correctly in `base.FS.move`.